frontend/simple/js/state.js

```diff
@@ -189,66 +221,74 @@ const actions = {
       }
-    } else {
-      return console.error(`UNKNOWN EVENT TYPE!`, contractId, entry)
-    }
 
-    // TODO: verify each entry is signed by a group member
-    // TODO: certainly we should have a larger try/catch block wrapping the entire function
-    await db.addLogEntry(contractId, entry)
+      // TODO: verify each entry is signed by a group member
+      // TODO: certainly we should have a larger try/catch block wrapping the entire function
+      await db.addLogEntry(contractId, entry)
 
-    if (contractId === hash) {
-      commit('addContract', {contractId, type, data: entry.toVuexState()})
-    }
-    commit('setRecentHash', {contractId, hash})
+      if (contractId === hash) {
+        commit('addContract', { contractId, type, data: entry.toVuexState() })
+      }
+      commit('setRecentHash', { contractId, hash })
 
-    // TODO: all of these might throw an exception -- handle those appropriately!
-    commit(`${contractId}/${type}`, {data: entry.data, hash})
-    if (store.state[contractId]._async.length) {
-      for (let asyncPayload of store.state[contractId]._async) {
-        await dispatch(`${contractId}/${asyncPayload.type}`, asyncPayload)
+      // TODO: all of these might throw an exception -- handle those appropriately!
+      commit(`${contractId}/${type}`, { data: entry.data, hash })
+      if (store.state[ contractId ]._async.length) {
+        for (let type of store.state[ contractId ]._async) {
+          await dispatch(`${contractId}/${type}`, { type, store, data: entry.data, hash })
+        }
+        commit(`${contractId}/clearAsync`)
+      }
+      if (contractId === state.currentGroupId) {
+        // this is to support EventLog.vue + TimeTravel.vue
+        // it's not super important and we'll probably get rid of it later
+        commit('setPosition', hash)
       }
-      commit(`${contractId}/clearAsync`)
-    }
-    if (contractId === state.currentGroupId) {
-      // this is to support EventLog.vue + TimeTravel.vue
-      // it's not super important and we'll probably get rid of it later
-      commit('setPosition', hash)
-    }
 
-    // handleEvent might be called very frequently, so save only after a pause
-    debouncedSave(dispatch)
-    // let any listening components know that we've received, processed, and stored the event
-    Vue.events.$emit(hash, contractId, entry)
-    Vue.events.$emit('eventHandled', contractId, entry)
+      // handleEvent might be called very frequently, so save only after a pause
+      await debouncedSave(dispatch, _.cloneDeep(store.state))
+      // let any listening components know that we've received, processed, and stored the event
+      Vue.events.$emit(hash, contractId, entry)
+      Vue.events.$emit('eventHandled', contractId, entry)
+      unlock()
+    } catch (ex) {
+      unlock()
+      throw ex
+    }
   },
   // persisting the state
-  async saveSettings (
-    {state}: {state: Object}
+  async saveSettings (vue,
+    state: {state: Object}
```

Comment:
This doesn't look right. State should be deconstructed as part of the first parameter, and not clear what this random `vue` is.